api/types/database.go

```diff
@@ -498,6 +498,13 @@ func (d *DatabaseV3) CheckAndSetDefaults() error {
 		return trace.BadParameter("MySQL ServerVersion can be only set for MySQL database")
 	}
 
+	// Validate AWS Specific configuration
+	if d.Spec.AWS.AccountID != "" {
+		if err := awsutils.IsValidAccountID(d.Spec.AWS.AccountID); err != nil {
+			return trace.BadParameter("invalid AWS Account ID: %v", err)
+		}
+	}
+
```

Comment:
Some of the URL parsers below extract account IDs from the URI, so I think we should validate this after the switch.

Reply:
:+1: